tests/test_roles.py

```diff
@@ -0,0 +1,757 @@
+#!/usr/bin/env python
+#
+# A library containing community-based extension for the python-telegram-bot library
+# Copyright (C) 2020
+# The ptbcontrib developers
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser Public License for more details.
+#
+# You should have received a copy of the GNU Lesser Public License
+# along with this program.  If not, see [http://www.gnu.org/licenses/].
+import datetime
+import os
+import pickle
+import subprocess
+
+import pytest
+import sys
+import time
+
+from copy import deepcopy
+from telegram import Message, User, InlineQuery, Update, ChatMember, Chat, TelegramError
+from telegram.ext import BasePersistence, CallbackContext, MessageHandler, Filters
+
+from ptbcontrib.roles import (
+    Role,
+    Roles,
+    ChatAdminsRole,
+    ChatCreatorRole,
+    setup_roles,
+    BOT_DATA_KEY,
+    RolesHandler,
+)
+
+
+@pytest.fixture(scope='module', autouse=True)
+def install_requirements():
+    subprocess.check_call(
+        [sys.executable, "-m", "pip", "install", "-r", "ptbcontrib/roles/requirements.txt"]
+    )
+
+
+@pytest.fixture(scope='function')
+def update():
+    return Update(
+        0,
+        Message(
+            0, datetime.datetime.utcnow(), Chat(0, 'private'), from_user=User(0, 'TestUser', False)
```

Comment:
I try to avoid that in order to not get confused between `datetime` and `datetime.datetime`. Changed to `import datetime as dtm` to abbreviate a bit …